extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -209,23 +217,38 @@ export class ResourceTypePickerDialog extends DialogBase {
 				if (this.toolRefreshTimestamp !== currentRefreshTimestamp) {
 					return;
 				}
+				let autoInstallRequired = false;
 				const messages: string[] = [];
 				this._toolsTable.data = toolRequirements.map(toolRef => {
 					const tool = this.toolsService.getToolByName(toolRef.name)!;
-					if (!tool.isInstalled) {
+					if (!tool.isInstalled && !tool.autoInstallSupported) {
 						messages.push(localize('deploymentDialog.ToolInformation', "{0}: {1}", tool.displayName, tool.homePage));
 						if (tool.statusDescription !== undefined) {
 							console.warn(localize('deploymentDialog.DetailToolStatusDescription', "Additional status information for tool: {0}. {1}", tool.name, tool.statusDescription));
 						}
 					}
+
+					if (!tool.isInstalled && tool.autoInstallSupported) {
+						autoInstallRequired = true;
+					}
 					return [tool.displayName, tool.description, tool.isInstalled ? localize('deploymentDialog.YesText', "Yes") : localize('deploymentDialog.NoText', "No"), tool.version ? tool.version.version : ''];
 				});
-				this._dialogObject.okButton.enabled = messages.length === 0;
+
+				this._installToolButton.hidden = !autoInstallRequired;
+				this._dialogObject.okButton.enabled = messages.length === 0 && !autoInstallRequired;
 				if (messages.length !== 0) {
 					messages.push(localize('deploymentDialog.VersionInformationDebugHint', "You will need to restart Azure Data Studio if the tools are installed after Azure Data Studio is launched to pick up the updated PATH environment variable. You may find additional details in the debug console."));
 					this._dialogObject.message = {
 						level: azdata.window.MessageLevel.Error,
-						text: localize('deploymentDialog.ToolCheckFailed', "Some required tools are not installed or do not meet the minimum version requirement."),
+						text: localize('deploymentDialog.ToolCheckFailed', "Some required tools are not installed."),
+						description: messages.join(EOL)
+					};
+				} else if (autoInstallRequired) {
+					// we don't have scenarioes that have mixed type of tools
+					// either we don't support auto install: docker, or we support auto install for all required tools
+					this._dialogObject.message = {
+						level: azdata.window.MessageLevel.Information,
+						text: localize('deploymentDialog.InstallToolsHint', "Some required tools are not installed, you can click the \"Install tools\" button to install them."),
 						description: messages.join(EOL)
 					};
```

Comment:
This isn't going to do anything since messages is empty.

Reply:
thanks for catching it.